apps/homescreen/js/app.js

```diff
@@ -540,6 +577,12 @@ const SETTINGS_VERSION = 0;
 
         this.pageHeight = pageHeight;
         this.pendingGridHeight = gridHeight;
+
+        if (!this.visualLoadComplete &&
+            Math.floor(visibleChildren / (this.small ? 4 : 3)) * iconHeight >=
+              scrollHeight) {
+          this.onVisualLoad();
+        }
```

Comment:
Indeed, this is what's happening - I think we're best fixing gaia-site-icon so that the fade animation it's meant to run actually works (somehow). I'll see if there's anyone on platform that knows the heuristics of image loading/painting and signals, but let's address this in follow-up. (note, this affects homescreen even without this patch, just this patch makes it appear *almost* correct, coincidentally :))